collector/kernel/buffered_poller.h

```diff
@@ -371,4 +377,8 @@ class BufferedPoller : public PerfPoller {
   DnsRequests dns_requests_;
 
   bool all_probes_loaded_;
+
+  std::shared_ptr<KernelCollectorRestarter> kernel_collector_restarter_;
+
+  std::atomic<bool> debug_bpf_lost_samples_ = false;
```

Comment:
Why aren't we encapsulating this in a more typical getter/setter pattern - rather than leaving a bare variable in the object?

Reply:
I followed the pattern of existing code (set_all_probes_loaded(), bool all_probes_loaded_) and of setter/getter use in previous companies, that a private member variable could/should be accessed directly from within the class and that a setter and/or getter should be provided if necessary to set and/or get from outside of the class.  In this case BufferedPoller::debug_bpf_lost_samples() is the setter and there is no need for a getter.

Are you suggesting something different?

Reply:
Too much time in Kotlin land, I suspect.  Really got into the habit of encapsulating all member access with this pattern.  So long as it is private and we're being consistent, than yes, you are correct - this shouldn't matter.